src/test/java/com/commercetools/sync/producttypes/utils/ProductTypesUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,129 @@
+import io.sphere.sdk.products.attributes.AttributeDefinitionBuilder;
+import io.sphere.sdk.products.attributes.AttributeDefinitionDraft;
+import io.sphere.sdk.products.attributes.AttributeDefinitionDraftBuilder;
+import io.sphere.sdk.products.attributes.StringAttributeType;
+import io.sphere.sdk.producttypes.ProductType;
+
+import static com.commercetools.sync.producttypes.utils.ProductTypesUpdateActionUtils.buildChangeDescriptionAction;
+import static com.commercetools.sync.producttypes.utils.ProductTypesUpdateActionUtils.buildSetKeyAction;
+import static com.commercetools.sync.producttypes.utils.ProductTypesUpdateActionUtils.buildChangeNameAction;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.sphere.sdk.producttypes.ProductTypeDraft;
+import io.sphere.sdk.producttypes.commands.updateactions.ChangeDescription;
+import io.sphere.sdk.producttypes.commands.updateactions.ChangeName;
+import io.sphere.sdk.producttypes.commands.updateactions.SetKey;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+public class ProductTypesUpdateActionUtilsTest {
+    private static ProductType old;
+    private static ProductTypeDraft newSame;
+    private static ProductTypeDraft newDifferent;
+    private static ProductTypeDraft newWithNullValues;
+
+    /**
+     * Initialises test data.
+     */
+    @BeforeClass
+    public static void setup() {
+        final LocalizedString label = LocalizedString.of(Locale.ENGLISH, "label1");
+        final AttributeDefinition attributeDefinition = AttributeDefinitionBuilder
+                .of("attributeName1", label, StringAttributeType.of())
+                .build();
+
+        final List<AttributeDefinition> oldAttributeDefinitions = singletonList(attributeDefinition);
+
+        final List<AttributeDefinitionDraft> sameAttributeDefinitionDrafts = singletonList(
+                AttributeDefinitionDraftBuilder.of(attributeDefinition).build());
+
+        old = mock(ProductType.class);
+        when(old.getKey()).thenReturn("key1");
+        when(old.getName()).thenReturn("name1");
+        when(old.getDescription()).thenReturn("description 1");
+        when(old.getAttributes()).thenReturn(oldAttributeDefinitions);
+
+        newSame = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "key1",
+                "name1",
+                "description 1",
+                sameAttributeDefinitionDrafts
+        );
+
+        newDifferent = ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "key2",
+                "name2",
+                "description 2",
+                sameAttributeDefinitionDrafts
+        );
+
+        newWithNullValues = ProductTypeDraft.ofAttributeDefinitionDrafts(
```

Comment:
maybe you missed to test cases with `null` values?

Reply:
I was about to test it but later I realized that name and description are required values and can't be null

I removed the unused variable